src/services/itemMembership/entities/ItemMembership.ts

```diff
@@ -26,14 +27,17 @@ export class ItemMembership extends BaseEntity implements GraaspItemMembership {
     nullable: false,
     enum: Object.values(PermissionLevel),
   })
+  @Index()
   permission: PermissionLevel;
 
   @ManyToOne(() => Member, (member) => member.id, {
     onDelete: 'SET NULL',
   })
+  @Index()
   @JoinColumn({ name: 'creator_id' })
   creator: Member;
```

Comment:
Not sure this one is necessary, I can't think of a case where we look for a membership given the creator.

Reply:
I was actually thinking of getOwn

```
async getOwn(memberId: string): Promise<Item[]> {
    return this.createQueryBuilder('item')
      .leftJoinAndSelect('item.creator', 'creator')
      .innerJoin('item_membership', 'im', 'im.item_path @> item.path')
      .where('creator.id = :id', { id: memberId })
      .andWhere('im.permission = :permission', { permission: PermissionLevel.Admin })
      .andWhere('nlevel(item.path) = 1')
      .orderBy('item.updatedAt', 'DESC')
      .getMany();
  },
```

But this is probably an index on item.creator then, you're right I will change it